src/main/java/gift/controller/ProductViewController.java

```diff
@@ -0,0 +1,60 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.service.ProductService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/products")
+public class ProductViewController {
+
+    private final ProductService productService;
+
+    public ProductViewController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    public String list(Model model) {
+        model.addAttribute("products", productService.findAll(0, 100, "id, asc"));
+        return "product/list";
+    }
```

Comment:
개인적으로는 컨트롤러 메소드의 파라미터인 Model 에 무언가가 주입되는 상황이 아니라면,
ModelAndView 와 같은 클래스의 객체를 만들어서 사용하는 편입니다. 

```suggestion
    @GetMapping
    public ModelAndView list() {
        Map<String, Object> model = new HashMap<>();
        model.put("products", productService.findAll(0, 100, "id, asc"));
        return new ModelAndView("product/list", model);
    }
```

위와 같은 코드가 될텐데요. 사용하는 변수의 스코프를 줄여 예상하지 못한 상황을 없애고, 서버사이드 렌더링을 위한 요소를 한 번에 다룬다는 점에서 선호하는 편입니다. 리턴타입이 String 이 아니라서 좀 더 명확하게 알려줄 수 있구요.
특별히 이 방식을 사용하시진 않아도 되고, 방법만 한 번 참고해주시면 좋겠습니다. 이미 알고 계실수도 있겠지만요 :)

Reply:
몰랐던 방법인데 좋은 것 같습니다! 존재하지 않는 상품에 대한 오류 페이지 반환도 이 방법을 채택해보겠습니다!